samza-core/src/main/java/org/apache/samza/container/grouper/stream/AllSspToSingleTaskGrouperFactory.java

```diff
@@ -19,51 +19,116 @@
+import org.apache.commons.lang3.Validate;
 import org.apache.samza.SamzaException;
 import org.apache.samza.config.Config;
+import org.apache.samza.config.ConfigException;
 import org.apache.samza.config.JobConfig;
+import org.apache.samza.config.JobCoordinatorConfig;
+import org.apache.samza.config.ZkConfig;
 import org.apache.samza.container.TaskName;
+import org.apache.samza.standalone.PassthroughJobCoordinatorFactory;
 import org.apache.samza.system.SystemStreamPartition;
+import org.apache.samza.zk.ZkJobCoordinatorFactory;
+import org.apache.samza.zk.ZkKeyBuilder;
+
+import static org.apache.samza.zk.ZkCoordinationUtilsFactory.*;
+import static org.apache.samza.zk.ZkJobCoordinatorFactory.*;
 
-import java.util.Collections;
-import java.util.Map;
-import java.util.Set;
 
 /**
- * AllSspToSingleTaskGrouper, as the name suggests, assigns all partitions to be consumed by a single TaskInstance
- * This is useful, in case of using load-balanced consumers like the new Kafka consumer, Samza doesn't control the
- * partitions being consumed by a task. Hence, it is assumed that there is only 1 task that processes all messages,
- * irrespective of which partition it belongs to.
- * This also implies that container and tasks are synonymous when this grouper is used. Taskname(s) has to be globally
- * unique within a given job.
+ * AllSspToSingleTaskGrouper creates TaskInstances equal to the number of containers and assigns all partitions to be
+ * consumed by each TaskInstance. This is useful, in case of using load-balanced consumers like the high-level Kafka
+ * consumer and Kinesis consumer, where Samza doesn't control the partitions being consumed by the task.
  *
- * Note: This grouper does not take in broadcast streams yet.
+ * Note:
+ * 1. This grouper does not take in broadcast streams yet.
+ * 2. This grouper is supported only with Yarn, Zk-standalone and Passthrough JobCoordinators.
  */
+
 class AllSspToSingleTaskGrouper implements SystemStreamPartitionGrouper {
+  private final boolean isPassthroughJobCoordinator;
+  private final int taskCount;
   private final int containerId;
 
-  public AllSspToSingleTaskGrouper(int containerId) {
+  AllSspToSingleTaskGrouper(boolean isPassthroughJobCoordinator, int containerId, int taskCount) {
+    this.isPassthroughJobCoordinator = isPassthroughJobCoordinator;
     this.containerId = containerId;
+    this.taskCount = taskCount;
   }
 
   @Override
   public Map<TaskName, Set<SystemStreamPartition>> group(final Set<SystemStreamPartition> ssps) {
+    Map<TaskName, Set<SystemStreamPartition>> groupedMap = new HashMap<>();
+
     if (ssps == null) {
       throw new SamzaException("ssp set cannot be null!");
     }
     if (ssps.size() == 0) {
       throw new SamzaException("Cannot process stream task with no input system stream partitions");
     }
 
-    final TaskName taskName = new TaskName(String.format("Task-%s", String.valueOf(containerId)));
+    if (isPassthroughJobCoordinator) {
```

Comment:
@prateekm  proposed that having a different grouper - `AllSspToAllTaskGrouper` would be cleaner for this scenario. That way, we don't muddle the existing `Grouper` implementation. I agree with @prateekm that it's a good idea! Thoughts?

Reply:
If we leave implementation part aside, why do we want to have a grouper which works for only one job coordinator and yet another grouper which is similar to the existing one but works with other coordinators ? As long as the behavior for passthru does not change, we should be ok, right ? Having too many groupers is also something that we want to avoid.